javascript/javascript3/week2/task4/script.js

```diff
@@ -0,0 +1,29 @@
+const url = 'https://api.covidtracking.com/v1/us/daily.json';
+
+//Fetching and waiting using promises and .then
+function getDataUsingPromises() {
+  setTimeout(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+      });
+  }, 3000);
+}
+
+getDataUsingPromises();
+
+//Fetching and waiting using async/await
+async function getDataUsingPromisesAsync() {
+  await new Promise((resolve) =>
+    setTimeout(() => {
+      resolve();
+    }, 3000)
+  );
+  const resData = await fetch(url);
+  const data = await resData.json();
+
+  console.log(data);
+}
+
+getDataUsingPromisesAsync();
```

Comment:
instead of nesting the fetch inside the Promise, could you create a timeoutPromise(), getDogsNames() and chain them like this:

```
timeoutPromise(3000)
.then(getDogsNames)
.then(res => {
    console.log(res);
})
```

does that make sense?

Reply:
If you mean the getDataUsingPromisesAsync() function, the task is to use "Do the 3 steps below using async/await"
Or I don't understand what you mean)

Reply:
Ok, that's fine. I will talk about it on Sunday during class :)